File: Facade/Facade.java
class SubSystem1 {
    public void method1() {
        System.out.println("this is method1 of SubSystem1");
    }
}

class SubSystem2 {
    public void method2() {
        System.out.println("this is method2 of SubSystem2");
    }
}

class SubSystem3 {
    public void method3() {
        System.out.println("this is method3 of SubSystem3");
    }
}

class Facade {
    private SubSystem1 subSystem1;
    private SubSystem2 subSystem2;
    private SubSystem3 subSystem3;
    
    private volatile Facade facade;
    private Facade() {
        subSystem1 = new SubSystem1();
        subSystem2 = new SubSystem2();
        subSystem3 = new SubSystem3();
    }

    public synchronized Facade getInstance() {
        if (facade == null)
            facade = new Facade();
        return facade;
    }

    public void method1() {
        subSystem1.method1();
    }

    public void method2() {
        subSystem2.method2();
    }

    public void method3() {
        subSystem3.method3();
    }

    public static void main(String[] args) {
        Facade facade = new Facade();
        facade.method2();
        facade.method1();
    }
}

File: Facade/facade.py
class Facade(object):
    def __init__(self, subSystem1, subSystem2, subSystem3):
        self.subSystem1 = subSystem1
        self.subSystem2 = subSystem2
        self.subSystem3 = subSystem3

    def method1(self):
        self.subSystem1.method1()

    def method2(self):
        self.subSystem2.method2()

    def method3(self):
        self.subSystem3.method3()


class SubSystem1(object):
    def method1(self):
        print("method1 in SubSystem1")


class SubSystem2(object):
    def method2(self):
        print("method2 in SubSystem2")


class SubSystem3(object):
    def method3(self):
        print("method3 in SubSystem3")


if __name__ == "__main__":
    facade = Facade(SubSystem1(), SubSystem2(), SubSystem3())
    facade.method1()
    facade.method2()
    facade.method3()
